jni/lib/src/lang/jboolean.dart

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) 2023, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+// Copyright (c) 2023, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+// ignore_for_file: overridden_fields
+
+import '../accessors.dart';
+import '../jobject.dart';
+import '../jni.dart';
+import '../third_party/generated_bindings.dart';
+import '../types.dart';
+
+class JBooleanType extends JObjType<JBoolean> {
+  const JBooleanType();
+
+  @override
+  String get signature => r"Ljava/lang/Boolean;";
+
+  @override
+  JBoolean fromRef(JObjectPtr ref) => JBoolean.fromRef(ref);
+
+  @override
+  JObjType get superType => const JObjectType();
+
+  @override
+  final superCount = 2;
+
+  @override
+  int get hashCode => (JBooleanType).hashCode;
+
+  @override
+  bool operator ==(Object other) {
+    return other.runtimeType == JBooleanType && other is JBooleanType;
+  }
+}
+
+class JBoolean extends JObject {
+  @override
+  late final JObjType<JBoolean> $type = type;
+
+  JBoolean.fromRef(
+    JObjectPtr ref,
+  ) : super.fromRef(ref);
+
+  /// The type which includes information such as the signature of this class.
+  static const type = JBooleanType();
+
+  static final _class = Jni.findJClass(r"java/lang/Boolean");
+
+  static final _ctorId =
+      Jni.accessors.getMethodIDOf(_class.reference, r"<init>", r"(Z)V");
+  JBoolean(bool boolean)
+      : super.fromRef(Jni.accessors.newObjectWithArgs(
+            _class.reference, _ctorId, [boolean ? 1 : 0]).object);
+
+  static final _falseId = Jni.accessors.getStaticFieldIDOf(
+    _class.reference,
+    r"FALSE",
+    r"Ljava/lang/Boolean;",
+  );
+  static JBoolean get falseValue => const JBooleanType().fromRef(Jni.accessors
+      .getStaticField(_class.reference, _falseId, JniCallType.objectType)
+      .object);
+
+  static final _trueId = Jni.accessors.getStaticFieldIDOf(
+    _class.reference,
+    r"TRUE",
+    r"Ljava/lang/Boolean;",
+  );
+  static JBoolean get trueValue => const JBooleanType().fromRef(Jni.accessors
```

Comment:
was this generated using jnigen? (Looking at `deleteOriginal` below I think not.)

I would be interested in an extension mechansim which lets us generate the class as usual, while adding our own extension methods in other files, then export our methods in from respective generated classes.

Reply:
It's a combination. The type classes were directly copied. The method returns were copied but the classes and method definitions (implementing Dart abstract classes) were manually written and changed.

Reply:
cc @dcharkes